plugins/lookup/etcd3.py

```diff
@@ -0,0 +1,195 @@
+            type: str
+        timeout:
+            description:
+            - Client timeout.
+            default: 60
+            env:
+            - name: ETCDCTL_DIAL_TIMEOUT
+            type: int
+        user:
+            description:
+            - Authentified user name.
+            env:
+            - name: ETCDCTL_USER
+            type: str
+        password:
+            description:
+            - Authentified user password.
+            env:
+            - name: ETCDCTL_PASSWORD
+            type: str
+        endpoints:
+            description:
+            - Counterpart of ETCDCTL_ENDPOINTS enviroment variable.
+              Specify the etcd3 connection with and URL form or <host>:<port> form
+            env:
+            - name: ETCDCTL_ENDPOINTS
+            type: str
+
+    requirements:
+    - "etcd3 >= 0.10"
+'''
+
+EXAMPLES = '''
+    - name: "a value from a locally running etcd"
+      debug:
+        msg: "{{ lookup('community.general.etcd3', 'foo/bar') }}"
+
+    - name: "values from multiple folders on a locally running etcd"
+      debug:
+        msg: "{{ lookup('community.general.etcd3', 'foo', 'bar', 'baz') }}"
+
+    - name: "look for a key prefix"
+      debug:
+        msg: "{{ lookup('community.general.etcd3', '/foo/bar', prefix=True) }}"
+
+    - name: "connect to etcd3 with a client certificate"
+      debug:
+        msg: "{{ lookup('community.general.etcd3', 'foo/bar', cert_cert='/etc/ssl/etcd/client.pem', cert_key='/etc/ssl/etcd/client.key') }}"
+'''
+
+RETURN = '''
+    _raw:
+        description:
+        - List of keys and associated values.
+        type: list
+        elements: complex
+'''
+
+import re
+import yaml
+
+from ansible.plugins.lookup import LookupBase
+from ansible.utils.display import Display
+from ansible.module_utils.basic import missing_required_lib
+from ansible.module_utils._text import to_native
+from ansible.plugins.lookup import LookupBase
+from ansible.errors import AnsibleLookupError
+
+try:
+    import etcd3
+    HAS_ETCD = True
+except ImportError:
+    HAS_ETCD = False
+
+display = Display()
+
+
+def etcd3_client(client_params):
+    try:
+        etcd = etcd3.client(**client_params)
+        etcd.status()
+    except Exception as exp:
+        raise AnsibleLookupError('Cannot connect to etcd cluster: %s' % (to_native(exp)))
+    return etcd
+
+
+class LookupModule(LookupBase):
+
+    def run(self, terms, variables, **kwargs):
+
+        self.set_options(var_options=variables, direct=kwargs)
+        ymldoc = yaml.safe_load(DOCUMENTATION)
+
+        if not HAS_ETCD:
+            display.error(missing_required_lib('etcd3'))
+            return None
+
+        # create the etcd3 connection parameters objet to pass to etcd3 class
+        # by copying options object and removing unneeded keys
+        client_params = dict(self._options)
+        for i in ('_terms', 'prefix', 'endpoints'):
+            if i in client_params:
+                client_params.pop(i)
```

Comment:
I would do this more explicitly, and more importantly by calling `get_option` instead of using `self._options` (which is a private implementation detail).

Reply:
I felt smarter to make a copy of the dict instead of looping to get parameters one by one. Is that really what you want ?